### DISCUSSION TITLE
Why do Power Platform Set Connection Variables require Application ID and Redirect URL?

### DISCUSSION THREAD

**Discussion by craig-lund-palm** [microsoft/powerplatform-build-tools#539] [Q&A]
When trying to use Set Connection variables and selecting Username/password authentication type an Application ID and Redirect URL are required. 
![image](https://github.com/microsoft/powerplatform-build-tools/assets/89538862/d09ab7bd-d495-45d4-afec-351c04b2ff47)

Why is this required and where is the documentation on these settings. See issue https://github.com/MicrosoftDocs/power-platform/issues/2155

**Comment by devkeydet** [comment]
@snizar007 and I had a quick discussion and I believe there are plans to improve the docs here. Until then, this doc has an explanation of why you need to create an app registration to authenticate to Dataverse.
https://learn.microsoft.com/en-us/power-apps/developer/data-platform/authenticate-oauth
The above doc also links to this walkthrough.
https://learn.microsoft.com/en-us/power-apps/developer/data-platform/walkthrough-register-app-azure-active-directory

HTH

**Comment by craig-lund-palm** [comment]
I understand App Registrations. If I had one, I would not need to use the Username/Password authentication method. This task is used to get the authentication details to use in a build variable. I am just trying to make sense of needing an "application id" and "redirect uri" when authenticating with a Username/Password.

**Reply by petrochuk** [reply]
Even when you connect with Username/Password you still connect as some kind of application. Think of it this way: this application is connecting on your behalf to Dataverse with provided username/password.

We also made it easier to create App registrations with [pac admin create-service-principal](https://learn.microsoft.com/en-us/power-platform/developer/cli/reference/admin#pac-admin-create-service-principal)

**Reply by craig-lund-palm** [reply]
I like that option... Using Visual Studio Code terminal I get this...

![image](https://github.com/microsoft/powerplatform-build-tools/assets/89538862/2f94a52f-6994-4025-bc2c-3a77d7aa61bd)

**Reply by snizar007** [reply]
Can you please share the pac-log.txt from your pac command error above?

**Reply by craig-lund-palm** [reply]
[pac-log.txt](https://github.com/microsoft/powerplatform-build-tools/files/12702827/pac-log.txt)

**Reply by craig-lund-palm** [reply]
First error was from auth profiles having been removed from device. Added a profile to the environment and specified the same environment on the next try. 

![image](https://github.com/microsoft/powerplatform-build-tools/assets/89538862/5efbc9ac-0987-40b9-bf02-12f4c24d2b1e)

Here is that log.
[pac-log.txt](https://github.com/microsoft/powerplatform-build-tools/files/12703362/pac-log.txt)
